cekit/generator/osbs.py

```diff
@@ -65,23 +67,47 @@ def prepare_artifacts(self):
         and fetches all of them
         """
 
-        logger.info("Handling artifacts...")
+        logger.info("Handling artifacts for OSBS...")
         target_dir = os.path.join(self.target, 'image')
         fetch_artifacts_url = []
 
         for image in self.images:
             for artifact in image.all_artifacts:
-                logger.info("Preparing artifact '{}'".format(artifact['name']))
-
-                if isinstance(artifact, _PlainResource) and \
-                        config.get('common', 'redhat'):
+                logger.info("Preparing artifact '{}' (of type {})".format(artifact['name'], type(artifact)))
+
+                if isinstance(artifact, _UrlResource):
+                    if 'md5' in artifact:
+                        md5value = artifact['md5']
+                    else:
```

Comment:
I'm not sure if I would do this. We should rely on the checksums provided by the user. In other case I could imagine two things could happen:

1. We calculate checksum for a bad artifact and pass it later to OSBS
2. Files to download could be very big.

I would just entirely skip handling URL artifacts which do not have checksums provided (required by OSBS `fetch-artifacts-url.yaml` file). I imagine these files (without checksum) should be handled similarly to what we do now (lookaside cache) since these do not fulfill all requirements to be handled differently.

Thoughts?

Side note - we should handle all checksums, not only md5, but md5, sha1, sha256 as well.

Reply:
Not sure. Looking at the descriptors, the checksums are optional with a urlartifact - so I think it makes sense to add handling for them. We only attempt to calculate if the URLArtifact hasn't got a checksum - which from cpaas they all should. 

> Side note - we should handle all checksums, not only md5, but md5, sha1, sha256 as well.

Agreed - to be updated.